atc/exec/set_pipeline_step.go

```diff
@@ -217,7 +207,16 @@ func (step *SetPipelineStep) run(ctx context.Context, state RunState) error {
 	}
 
 	fmt.Fprintf(stdout, "setting pipeline: %s\n", step.plan.Name)
-	pipeline, _, err = parentBuild.SavePipeline(step.plan.Name, atcConfig, fromVersion, false)
+	parentBuild, found, err := step.buildFactory.Build(step.metadata.BuildID)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		return fmt.Errorf("set_pipeline step not attached to a buildID")
+	}
+
+	pipeline, _, err = parentBuild.SavePipeline(step.plan.Name, team.ID(), atcConfig, fromVersion, false)
 	if err != nil {
 		return err
```

Comment:
Previously, we discussed special casing on `db.ErrSetByNewerBuild` here to log a warning but not error the build outright. Is that still something we want to do? e.g.

```go
	if err != nil {
		if err == db.ErrSetByNewerBuild {
			fmt.Fprintln(stderr, "\x1b[1;33mWARNING: the pipeline was set by a newer build, so this step is a no-op\x1b[0m")
			return nil
		}
		return err
	}
```